lib/livebook/notebook/explore/intro_to_nx.livemd

```diff
@@ -1,3 +1,732 @@
+
+The new tensor has the same type, but a new shape.
+
+<!-- livebook:{"break_markdown":true} -->
+
+Now, reshape the tensor to contain three dimensions with
+one batch, one row, and four columns.
+
+```elixir
+# ...your code here...
+```
+
+We can create a tensor with named dimensions, a type, a shape,
+and our target data. A dimension is called an _axis_, and axes
+can have names. We can specify the tensor type and dimension names
+with options, like this:
+
+```elixir
+Nx.tensor([[1, 2, 3]], names: [:rows, :cols], type: {:u, 8})
+```
+
+We created a tensor of the shape `{1, 3}`, with the type `u8`,
+the values `[1, 2, 3]`, and two axes named `rows` and `cols`.
+
+<!-- livebook:{"break_markdown":true} -->
+
+Now we know how to create tensors, so it's time to do something with them.
+
+## Tensor aware functions
+
+In the last section, we created a `s64[2][2]` tensor. In this section,
+we'll use Nx functions to work with it. Here's the value of `tensor`:
+
+```elixir
+tensor
+```
+
+We can use `IEx.exports/1` or code completion to find
+some functions in the `Nx` module that operate on tensors:
+
+```elixir
+exports(Nx)
+```
+
+You might recognize that many of those functions have names that
+suggest that they would work on primitive values, called scalars.
+Indeed, a tensor can be a scalar:
+
+```elixir
+pi = Nx.tensor(3.1415, type: {:f, 32})
+```
+
+Take the cosine:
+
+```elixir
+Nx.cos(pi)
+```
+
+That function took the cosine of `pi`.  We can also call them 
+on a whole tensor, like this:
+
+```elixir
+Nx.cos(tensor)
+```
+
+We can also call a function that aggregates the contents
+of a tensor. For example, to get a sum of the numbers
+in `tensor`, we can do this:
+
+```elixir
+Nx.sum(tensor)
+```
+
+That's `1 + 2 + 3 + 4`, and Nx went to multiple dimensions to get that sum.
+To get the sum of values along the `x` axis instead, we'd do this:
+
+```elixir
+Nx.sum(tensor, axes: [:x])
+```
+
+Nx sums the values across the `x` dimension: `1 + 2` in the first row
+and `3 + 4` in the second row.
+
+<!-- livebook:{"break_markdown":true} -->
+
+Now,
+
+* create a `{2, 2, 2}` tensor
+* with the values `1..8`
+* with dimension names `[:z, :y, :x]`
+* calculate the sums along the `y` axis
+
+```elixir
+# ...your code here...
+```
+
+Sometimes, we need to combine two tensors together with an
+operator. Let's say we wanted to add two tensors together.
+Mathematically, the expression looks like this:
+
+<!-- livebook:{"break_markdown":true} -->
+
+$$
+\begin{bmatrix}
+  5 & 6 \\
+  7 & 8
+\end{bmatrix}
+
+-
+\begin{bmatrix}
+  1 & 2 \\
+  3 & 4
+\end{bmatrix}
+
+\begin{bmatrix}
+  4 & 4 \\
+  4 & 4
+\end{bmatrix}
+$$
+
+<!-- livebook:{"break_markdown":true} -->
+
+To solve this problem, add each integer on the left with the
+corresponding integer on the right. In Nx, this operation
+is called tensor-aware addition. Let's use the `+` function
+to solve the problem:
+
+```elixir
+tensor2 = Nx.tensor([[5, 6], [7, 8]])
+Nx.Defn.Kernel.-(tensor2, tensor)
```

Comment:
I wouldn't recommend people to call `Nx.Defn.Kernel` directly. So instead, I would use Nx.add/Nx.subtract and then still mention it is tedious and say we would like to use a mathematical notation. :)